integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/iceberg/SparkIcebergCatalogIT.java

```diff
@@ -126,4 +143,145 @@ void testIcebergFunction() {
+        String.format(
+            "INSERT into %s values(2,'a',cast('2024-01-01 12:00:00.0' as timestamp));", tableName);
+    sql(insertData);
+    List<String> queryResult = getTableData(tableName);
+    Assertions.assertEquals(1, queryResult.size());
+    Assertions.assertEquals("2,a,2024-01-01 12:00:00.0", queryResult.get(0));
+    String partitionExpression = "ts_month=2024-01";
+    Path partitionPath = new Path(getTableLocation(tableInfo), partitionExpression);
+    checkDirExists(partitionPath);
+  }
+
+  @Test
+  void testCreateIcebergYearPartitionTable() {
+    String tableName = "iceberg_year_partition_table";
+    dropTableIfExists(tableName);
+    String createTableSQL = getCreateIcebergSimpleTableString(tableName);
+    createTableSQL = createTableSQL + " PARTITIONED BY (years(ts));";
+    sql(createTableSQL);
+    SparkTableInfo tableInfo = getTableInfo(tableName);
+    SparkTableInfoChecker checker =
+        SparkTableInfoChecker.create()
+            .withName(tableName)
+            .withColumns(getIcebergSimpleTableColumn())
+            .withYearPartition(Collections.singletonList("ts"));
+    checker.check(tableInfo);
+
+    String insertData =
+        String.format(
+            "INSERT into %s values(2,'a',cast('2024-01-01 12:00:00.0' as timestamp));", tableName);
+    sql(insertData);
+    List<String> queryResult = getTableData(tableName);
+    Assertions.assertEquals(1, queryResult.size());
+    Assertions.assertEquals("2,a,2024-01-01 12:00:00.0", queryResult.get(0));
+    String partitionExpression = "ts_year=2024";
+    Path partitionPath = new Path(getTableLocation(tableInfo), partitionExpression);
+    checkDirExists(partitionPath);
+  }
+
+  @Test
+  void testCreateIcebergTruncatePartitionTable() {
+    String tableName = "iceberg_truncate_partition_table";
+    dropTableIfExists(tableName);
+    String createTableSQL = getCreateIcebergSimpleTableString(tableName);
+    createTableSQL = createTableSQL + " PARTITIONED BY (truncate(1, name));";
+    sql(createTableSQL);
+    SparkTableInfo tableInfo = getTableInfo(tableName);
+    SparkTableInfoChecker checker =
+        SparkTableInfoChecker.create()
+            .withName(tableName)
+            .withColumns(getIcebergSimpleTableColumn())
+            .withTruncatePartition(1, Collections.singletonList("name"));
+    checker.check(tableInfo);
+
+    String insertData =
+        String.format(
+            "INSERT into %s values(2,'a',cast('2024-01-01 12:00:00.0' as timestamp));", tableName);
+    sql(insertData);
+    List<String> queryResult = getTableData(tableName);
+    Assertions.assertEquals(1, queryResult.size());
+    Assertions.assertEquals("2,a,2024-01-01 12:00:00.0", queryResult.get(0));
+    String partitionExpression = "name_trunc=a";
+    Path partitionPath = new Path(getTableLocation(tableInfo), partitionExpression);
+    checkDirExists(partitionPath);
+  }
+
+  private String getCreateIcebergSimpleTableString(String tableName) {
```

Comment:
please add description about why not add a new method to generate create table string not using `getCreateSimpleTableString`

Reply:
done.